server/src/testFixtures/java/io/crate/testing/TestingHelpers.java

```diff
@@ -436,4 +438,18 @@ public static void assertCrateVersion(Object object, Version versionCreated, Ver
                     is(Version.Property.UPGRADED.toString()),
                     versionUpgraded == null ? nullValue() : is(versionUpgraded.externalNumber()))));
     }
+
+    public static <T> List<T> getRandomsOfType(int minLength, int maxLength, DataType<T> dataType) {
+        var values = new ArrayList<T>();
+        if(minLength < 2) {
+            throw new IllegalArgumentException("minLength must be greater than 1. One item is reserved for NULL and there must be at least 1 not NULL value.");
+        }
+        int length  = RandomizedTest.randomIntBetween(minLength, maxLength);
+        var generator = DataTypeTesting.getDataGenerator(dataType);
+        for (int i = 0; i <= length - 1; i++) {
+            values.add(dataType.sanitizeValue(generator.get()));
+        }
+        values.add(null); //One NULL value is always included.
```

Comment:
I think it might make sense to do the `.add(null)` on a random basis as well, to ensure arrays without any null values are also tested.

And I think it shouldn't be added as an additional element. That the list that is returned can exceeds `maxLength` could be surprising.

Reply:
loop is to `length - 1` and if  `maxLength<2` Exception with "..One item is reserved for NULL." is thrown  - null case can be random indeed, will add that, thanks